Logging issues with log4j2-jboss-logmanager

**Describe the bug**
When using the **log4j2-jboss-logmanager**, logs are not printed when running tests.
Furthermore, when using a log4j logger next to a jboss logger, a stack trace is reported "Maximum number of attachments exceeded".

**Expected behavior**
I would expect logs to be printed during tests.
I would also expect the runner jar to start when using log4j + jboss loggers.

**To Reproduce**

A reproducer could be found at https://github.com/aldettinger/quarkus-log4j-issue-reproducer.
Steps to reproduce the behavior:
1. mvn clean package
2. => NO logs are issued to console during tests (while they are when using the runner jar)
3. Uncomment the two jboss logger related lines in HelloResource.java
4. mvn clean package
5. Still no logs are issued during tests
6. Moreover the stack trace below is issued:
```
Exception in thread "main" java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.lang.Class.newInstance(Class.java:584)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:60)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:189)
        ... 9 more
Caused by: java.lang.RuntimeException: Failed to initialize Arc
        at io.quarkus.arc.Arc.initialize(Arc.java:26)
        at io.quarkus.arc.runtime.ArcRecorder.getContainer(ArcRecorder.java:24)
        at io.quarkus.deployment.steps.ArcProcessor$generateResources-1025303321.deploy_0(ArcProcessor$generateResources-1025303321.zig:76)
        at io.quarkus.deployment.steps.ArcProcessor$generateResources-1025303321.deploy(ArcProcessor$generateResources-1025303321.zig:40)
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:147)
        ... 9 more
Caused by: java.lang.ExceptionInInitializerError
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:398)
        at io.quarkus.reproducer.HelloResource_Bean.<init>(HelloResource_Bean.zig:40)
        at io.quarkus.arc.setup.Default_ComponentsProvider.addBeans1(Default_ComponentsProvider.zig:134)
        at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Default_ComponentsProvider.zig:38)
        at io.quarkus.arc.impl.ArcContainerImpl.<init>(ArcContainerImpl.java:113)
        at io.quarkus.arc.Arc.initialize(Arc.java:20)
        ... 13 more
Caused by: java.lang.IllegalStateException: Maximum number of attachments exceeded
        at org.jboss.logmanager.LoggerNode.attachIfAbsent(LoggerNode.java:428)
        at org.jboss.logmanager.Logger.attachIfAbsent(Logger.java:194)
        at org.jboss.logging.JBossLogManagerProvider.doGetLogger(JBossLogManagerProvider.java:93)
        at org.jboss.logging.JBossLogManagerProvider.getLogger(JBossLogManagerProvider.java:53)
        at org.jboss.logging.Logger.getLogger(Logger.java:2465)
        at org.jboss.logging.Logger.getLogger(Logger.java:2490)
        at io.quarkus.reproducer.HelloResource.<clinit>(HelloResource.java:24)
```
**Environment (please complete the following information):**
 - Output of `uname -a` ~=
```
   Linux ag.remote.csb 3.10.0-957.27.2.el7.x86_64 #1 SMP Tue Jul 9 16:53:14 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
 - Output of `java -version`: 
```
   openjdk version "11.0.8" 2020-07-14
   OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
   OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)
```
 - Quarkus version or git rev:
```
   1.9.0.CR1
```
 - Build tool mvn -version: 
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/ag/dev/maven/apache-maven-3.6.3
Java version: 11.0.8, vendor: AdoptOpenJDK, runtime: /home/ag/.sdkman/candidates/java/11.0.8.hs-adpt
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.10.0-957.27.2.el7.x86_64", arch: "amd64", family: "unix"
```
**Additional context**
The global context is that I'm experimenting around an extension that would involve 3rd party libraries making use of distinct kind of loggers. I tried to shrink the reproducer as far as possible.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -153,7 +153,7 @@
         <maven-invoker.version>3.0.1</maven-invoker.version>
         <awaitility.version>4.0.3</awaitility.version>
         <jprocesses.version>1.6.5</jprocesses.version>
-        <jboss-logmanager.version>1.0.4</jboss-logmanager.version>
+        <jboss-logmanager.version>1.0.5</jboss-logmanager.version>
         <jgit.version>5.8.0.202006091008-r</jgit.version>
         <flyway.version>6.5.7</flyway.version>
         <yasson.version>1.0.8</yasson.version>

